apps/ledger-live-desktop/src/renderer/modals/CreateLocalManifest/index.tsx

```diff
@@ -0,0 +1,364 @@
+  categories: ["NFT", "Swap", "YourAppCategory"],
+  currencies: ["*"],
+  highlight: false,
+  content: {
+    cta: { en: "" },
+    subtitle: { en: "" },
+    shortDescription: {
+      en: "shortDescription",
+    },
+    description: {
+      en: "description",
+    },
+  },
+  domains: ["http://*"],
+  visibility: "complete",
+  permissions: [
+    "account.list",
+    "account.receive",
+    "account.request",
+    "currency.list",
+    "device.close",
+    "device.exchange",
+    "device.transport",
+    "message.sign",
+    "transaction.sign",
+    "transaction.signAndBroadcast",
+    "wallet.capabilities",
+    "wallet.info",
+    "wallet.userId",
+    "storage.get",
+    "storage.set",
+  ],
+  dapp: {
+    provider: "evm",
+    nanoApp: "nanoApp",
+    networks: [
+      {
+        currency: "ethereum",
+        chainID: 1,
+        nodeURL: "https://eth-dapps.api.live.ledger.com",
+      },
+    ],
+  },
+};
+
+function createLocalManifest() {
+  return (
+    <Modal
+      name="MODAL_CREATE_LOCAL_APP"
+      centered
+      render={({ data, onClose }) => <FormLocalManifest data={data} onClose={onClose} />}
+    />
+  );
+}
+
+function FormLocalManifest({
+  data,
+  onClose,
+}: {
+  data: { manifest?: LiveAppManifest };
+  onClose: () => void;
+}) {
+  const { addLocalManifest } = useLocalLiveAppContext();
+  const [form, setForm] = useState<LiveAppManifest>({ ...DEFAULT_FORM });
+  const [isDapp, setIsDapp] = useState("dapp" in form);
+
+  const handleSwitchEthDapp = () => {
+    setIsDapp(prevState => !prevState);
+    setForm(prevState => {
+      !isDapp
+        ? (prevState.dapp = { ...DEFAULT_FORM.dapp } as LiveAppManifestDapp)
+        : delete prevState.dapp;
+      return prevState;
+    });
+  };
+
+  const formIsValid = useMemo(() => LiveAppManifestSchema.safeParse(form).success, [form]);
+
+  const handleChange = (path: string, value: unknown) => {
+    setForm((prevState: LiveAppManifest) => {
+      return updateObjectAtPath({ ...prevState }, path, value);
+    });
+  };
+
+  return (
+    <form onSubmit={() => formIsValid && addLocalManifest(form)}>
+      <ModalBody
+        onClose={onClose}
+        onBack={undefined}
+        title={
+          <Trans
+            i18nKey={`settings.developer.createLocalAppModal.title.${
+              data.manifest ? "modify" : "create"
+            }`}
+          />
+        }
+        noScroll
+        render={() => (
+          <>
+            <ScrollArea>
+              <Flex height={"900px"} rowGap={10} flexDirection={"column"}>
+                <Flex marginY={"10px"} columnGap={2}>
+                  <Text marginBottom={1} marginLeft={1} ff="Inter|Medium" fontSize={4}>
+                    {`ETH Dapp`}
+                  </Text>
+                  <Switch isChecked={isDapp} onChange={handleSwitchEthDapp} />
+                  {isDapp && (
+                    <Text paddingTop={1} ff="Inter|Medium" fontSize={2}>
+                      {`dapp field added`}
```

Comment:
is this for debugging purposes?   
It's unclear for the user until you go at the end of the form.

Reply:
i can modify it to, dapp field added at the end